File: ARRAYS/1. Two Sum.java
//problem link: https://leetcode.com/problems/two-sum/?envType=problem-list-v2&envId=array
//timeComplexity: O(n)
//spaceCompelxity: O(n)

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer>freq=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            freq.put(nums[i],i);
        }
        for(int i=0;i<nums.length;i++){
            int val=target-nums[i];
            if(freq.containsKey(val) && i!=freq.get(val)){
                return new int[]{i,freq.get(val)};
            }
        }
        return new int[]{-1,-1};
    }
}

File: ARRAYS/1. Two Sum.cpp
//problem link: https://leetcode.com/problems/two-sum/
//timeComplexity: o(n)
//spaceComplexity: O(n)

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int>freq;
        for(int i=0;i<n;i++){
            int val=target-nums[i];
            if(freq.find(val)!=freq.end()){
                return {i,freq[val]};
            }
            freq[nums[i]]=i;
        }
        return {-1,-1};
    }
};
